flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceSinkFactoryBase.java

```diff
@@ -322,6 +333,26 @@ private StartupOptions getStartupOptions(
 		return options;
 	}
 
+	private void buildSpecificOffsets(DescriptorProperties descriptorProperties, String topic, Map<KafkaTopicPartition, Long> specificOffsets) {
+		if (descriptorProperties.containsKey(CONNECTOR_SPECIFIC_OFFSETS)) {
+			final Map<Integer, Long> offsetMap = KafkaValidator.validateAndGetSpecificOffsetsStr(descriptorProperties);
+			offsetMap.forEach((partition, offset) -> {
+				final KafkaTopicPartition topicPartition = new KafkaTopicPartition(topic, partition);
+				specificOffsets.put(topicPartition, offset);
+			});
+		} else {
+			final List<Map<String, String>> offsetList = descriptorProperties.getFixedIndexedProperties(
+					CONNECTOR_SPECIFIC_OFFSETS,
+					Arrays.asList(CONNECTOR_SPECIFIC_OFFSETS_PARTITION, CONNECTOR_SPECIFIC_OFFSETS_OFFSET));
+			offsetList.forEach(kv -> {
+				final int partition = descriptorProperties.getInt(kv.get(CONNECTOR_SPECIFIC_OFFSETS_PARTITION));
```

Comment:
Don't need every field contains `final`.

Reply:
this pice come from original code and I think we'd better to keep final